### ISSUE TITLE
Field added to section but Inventory not reporting it

### ISSUE THREAD

**Issue by sg-modlab** [lindseydiloreto/craft-inventory#17]
Exploring Inventory to help with a site redesign and build but confidence in plugin is not yet strong. Wondering if there is something I need to do.

I have at least one case where a matrix field is added to a section but that field is not being shown by inventory as being used anywhere. Is there a setting or cache or something that might be need to be set, cleared, or something done to make sure getting accurate results? 

Does Inventory support all Craft basic field types? Does it support other plugin fields?

This seems like it could be a very helpful plugin to clean up unused fields and modify others when trying to simplify a site's architecture. 

Thanks,
Seth

**Comment by lindseydiloreto** [comment]
Thanks for reporting, sorry for the late reply! I've heard whispers of this bug, and I think I've even seen it happen once.

But I'm having trouble reproducing it in my local sandbox...

![can't reproduce](https://user-images.githubusercontent.com/5309692/222761831-021ded48-0167-43a0-8017-704261262f04.png)

@sg-modlab I'll hit you up on Discord, we'll see if we can squash the bug there. 👍

**Comment by lindseydiloreto** [comment]
For anyone following along, we weren't able to reproduce this.

If you are running into this issue and are willing to share your database, please track me down on Craft Discord. 🙏

**Comment by thisisjamessmith** [comment]
I'm having the same issue. Not matrix fields in particular, but I've spotted it so far on a Redactor field and and Checkbox Group field. The site in question has been upgraded from Craft 3 to 4. Inventory was only installed after the upgrade. The only way I can get accurate results in Inventory is by making a change in the section's field layout and resaving it. (Resaving without making a change isn't good enough). Interestingly, after reverting the change, Git shows no changes to the project config files (other than the date modified), but clearly something in the database must have changed!

**Comment by lindseydiloreto** [comment]
@thisisjamessmith Thanks for reporting. Would you be able to share a copy of your database with me, so that I can reproduce the issue? We can move this to Discord if you'd like.

**Comment by lindseydiloreto** [comment]
Great news, I believe I have it all worked out! If possible, please update to the latest dev version to see if it's fixed...

```json
"doublesecretagency/craft-inventory": "dev-v3-dev"
```

**TL;DR** - The `fieldlayoutfields` table is borderline obsolete, and the `fieldlayouttabs` table is where we should be looking for field + layout relationships.

---

**Backstory** - Inventory was first released in 2016, during the heyday of Craft 2. It's since been updated for C3 and C4.

At some point (probably when the **field layout UX** got a huge makeover), all of the field relationship data was moved into `fieldlayouttabs`. This allowed the layout UX to be much more flexible... but Inventory was still watching the `fieldlayoutfields` table.

When those two tables are in sync, the problem is not noticeable.

At a subsequent point, there was a Craft migration (between 3.x -> 4.x) which left a gap of data in the `fieldlayoutfields` table. Perhaps it was an oversight or deemed unnecessary, since Craft now looks at the `fieldlayouttabs` table almost exclusively.

When the tables became out of sync, Inventory reported those fields as "not in use".

**Once we figured all of this out,** it was simply a matter of rewriting the SQL query to rely on the `fieldlayouttabs` table.

---

Shout out to @olivierbon of Pixel + Tonic for helping me crack the code! 🍺

**Comment by thisisjamessmith** [comment]
This is working great now, many thanks Lindsey.

**Comment by lindseydiloreto** [comment]
Awesome, thanks for reporting back @thisisjamessmith! This fix has been officially released in [v3.1.0](https://github.com/doublesecretagency/craft-inventory/releases/tag/3.1.0). 👍
